dotCMS/src/main/java/com/dotcms/contenttype/business/UniqueFieldFactoryImpl.java

```diff
@@ -0,0 +1,24 @@
+package com.dotcms.contenttype.business;
+
+import com.dotmarketing.common.db.DotConnect;
+import com.dotmarketing.exception.DotDataException;
+
+import java.util.Map;
+
+/**
+ * Default implementation of {@link UniqueFieldFactory}
+ */
+public class UniqueFieldFactoryImpl implements UniqueFieldFactory {
+
+    private final static String INSERT_SQL = "INSERT INTO unique_fields (unique_key_val, supporting_values) VALUES (?, ?)";
+    /**
+     * Default implementation of {@link UniqueFieldFactory#insert(String, Map)}
+     *
+     * @param key
+     * @param supportingValues
+     */
+    @Override
+    public void insert(final String key, final Map<String, Object> supportingValues) throws DotDataException {
```

Comment:
the factory concept usually is next to an API, since you do not have any, and there are not any plans of covering more general cases, this is just a single line you may move straight to the Helper/API you have